orchagent/switchorch.cpp

```diff
@@ -49,6 +55,29 @@ SwitchOrch::SwitchOrch(DBConnector *db, vector<TableConnector>& connectors, Tabl
     querySwitchTpidCapability();
     auto executorT = new ExecutableTimer(m_sensorsPollerTimer, this, "ASIC_SENSORS_POLL_TIMER");
     Orch::addExecutor(executorT);
+
+    vector<sai_attribute_t> attrs;
+    sai_attribute_t attr;
+    sai_status_t status;
+
+    attr.id = SAI_SWITCH_TUNNEL_ATTR_TUNNEL_TYPE;
+    attr.value.s32 = SAI_TUNNEL_TYPE_VXLAN;
+    attrs.push_back(attr);
+    attr.id = SAI_SWITCH_TUNNEL_ATTR_TUNNEL_VXLAN_UDP_SPORT_MODE;
+    attr.value.s32 = SAI_TUNNEL_VXLAN_UDP_SPORT_MODE_EPHEMERAL;
```

Comment:
This is not required as the default is "EPHEMERAL". Moreover, it might be possible for SAI to return a failure if this attribute is not supported. So I don't really find a reason for setting it in application.

Reply:
If  "EPHEMERAL" is default attr so SAI must support it anyway isn't it? Please correct me if I wrong. 

Anyway you proposal is to move **create_switch_tunnel(&switch_tunnel_id,** from constructor to **setSwitchTunnelVxlanParams** function?

Reply:
Thats not true. Attribute set support may or may not be present for all SAI implementation. I don't think we need to explicitly set default value here.